src/Microsoft.AspNetCore.Razor.Language/Syntax/MarkupElementRewriter.cs

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Language.Syntax
+{
+    internal static class MarkupElementRewriter
```

Comment:
>MarkupElementRewriter [](start = 26, length = 21)

I'm assuming that this is also temporary (this is the code that groups elements/tags).

Reply:
Yes but I didn't put it in Legacy because this operates completely on the new infrastructure and doesn't use any of out legacy components.

---
In reply to: [233619189](https://github.com/aspnet/Razor/pull/2590#discussion_r233619189) [](ancestors = 233619189)